specification/network/resource-manager/Microsoft.Network/stable/2024-03-01/azureFirewall.json

```diff
@@ -0,0 +1,1350 @@
+          "description": "Number of packets to be captured."
+        },
+        "sasUrl": {
+          "type": "string",
+          "description": "Upload capture location"
+        },
+        "fileName": {
+          "type": "string",
+          "description": "Name of file to be uploaded to sasURL"
+        },
+        "protocol": {
+          "$ref": "#/definitions/AzureFirewallNetworkRuleProtocol",
+          "description": "The protocol of packets to capture"
+        },
+        "flags": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AzureFirewallPacketCaptureFlags"
+          },
+          "description": "The tcp-flag type to be captured. Used with protocol TCP",
+          "x-ms-identifiers": []
+        },
+        "filters": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AzureFirewallPacketCaptureRule"
+          },
+          "description": "Rules to filter packet captures.",
+          "x-ms-identifiers": []
+        }
+      },
+      "description": "Azure Firewall Packet Capture Parameters."
+    },
+    "AzureFirewallPacketCaptureFlags": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "description": "Flags to capture",
+          "$ref": "#/definitions/AzureFirewallPacketCaptureFlagsType"
+        }
+      },
+      "description": "Properties of the AzureFirewallRCAction."
+    },
+    "AzureFirewallPacketCaptureFlagsType": {
+      "type": "string",
+      "description": "The flags type to be captured.",
+      "enum": [
+        "fin",
+        "syn",
+        "rst",
+        "push",
+        "ack",
+        "urg"
+      ],
+      "x-ms-enum": {
+        "name": "AzureFirewallPacketCaptureFlagsType",
+        "modelAsString": true
+      }
+    },
+    "AzureFirewallPacketCaptureRule": {
+      "type": "object",
+      "properties": {
+        "sources": {
+          "type": "array",
+          "description": "List of source IP addresses/subnets to be captured.",
+          "items": {
+            "type": "string"
+          }
+        },
+        "destinations": {
+          "type": "array",
+          "description": "List of destination IP addresses/subnets to be captured.",
+          "items": {
+            "type": "string"
+          }
+        },
+        "destinationPorts": {
+          "type": "array",
+          "description": "List of ports to be captured.",
+          "items": {
+            "type": "string"
+          }
+        }
+      },
+      "description": "Group of src/dest ips and ports to be captured."
+    },
+    "AzureFirewallAutoscaleConfiguration": {
+      "type": "object",
+      "properties": {
+        "minCapacity": {
+          "type": "integer",
+          "format": "int32",
+          "minimum": 2,
+          "description": "The minimum number of capacity units for this azure firewall. Use null to reset the value to the service default.",
+          "x-nullable": true
```

Comment:
>   "x-nullable": true [](http://example.com/codeflow?start=7&length=21)

x-nullable, or representation with explicit nulls, is not generally recommended in Azure APIs.
Is there a reason you need it??

Reply:
Following changes were introduced by PR : https://github.com/Azure/azure-rest-api-specs/pull/29933/files#diff-451f3487bb10debee2bd5e66d2483e6c86e72c7d134b4431bb027cf0b9ceec38 which is already approved